packages/layout/src/helpers/css-variables.ts

```diff
@@ -0,0 +1,83 @@
+import {
+  BREAKPOINTS,
+  Responsive,
+  getValueForEachBreakpoint,
+  isResponsiveObject,
+} from "./breakpoints";
+import { deriveSpacing } from "./spacing";
+
+type CssVariableProp = { rule: string; type: string };
+type ProcessCssVariableStringArgs = {
+  prop: CssVariableProp;
+  key: string;
+  value: string;
+  cssVariables: Record<string, string>;
+};
+
+const processCssVariableString = ({
+  prop,
+  value,
+  cssVariables,
+}: ProcessCssVariableStringArgs) => {
+  if (prop.type.includes("spacing")) {
+    const currentValueOfCssVariable = cssVariables[`--tgph-${prop.rule}`] || "";
+    return deriveSpacing({
+      value,
+      type: prop.type,
+      currentValue: currentValueOfCssVariable,
+    });
+  }
+
+  return value;
+};
+
+type PropToCssVariablesArgs = {
+  props: Record<string, Responsive<string>>;
+  ref: React.RefObject<HTMLElement>;
+  propsMap: Record<string, CssVariableProp>;
```

Comment:
Yeah, what do you think would be a more straightforward name? `props` are the actual props passed into the component where as `propsMap` is the mapping of prop info in the constants file.

Reply:
Ah I seeee. Maybe `props` and `mappings`? Or maybe i'm overthinking here, because I initially thought the "map" in "propsMap" was referring to the data type i.e. Record, but now I see it's referring to the mapping. 

The name of this arg isn't super consequential at the end of the day, so I will defer the final call to you. I think fine to leave it be if we can't think of a better name.